vllm/v1/engine/llm_engine.py

```diff
@@ -129,7 +135,44 @@ def add_request(
         prompt_adapter_request: Optional[PromptAdapterRequest] = None,
         priority: int = 0,
     ) -> None:
-
+        """Add request."""
+        # Handle parallel sampling requests differently.
+        if params is None or isinstance(params,
+                                        PoolingParams) or params.n == 1:
+            self._add_request(request_id=request_id,
+                              prompt=prompt,
+                              params=params,
+                              arrival_time=arrival_time,
+                              lora_request=lora_request,
+                              trace_headers=trace_headers,
+                              prompt_adapter_request=prompt_adapter_request,
+                              priority=priority)
```

Comment:
to simplify the code here, could make a dict of the common kwargs and use that in both the if and the else with **kwargs
```python
kwargs = dict(
    request_id=request_id,
    params=params,
    # ...
```